src/proxy.c

```diff
@@ -277,14 +279,14 @@ cc_proxy_msg_read (GIOChannel *source, GIOCondition condition,
 				*proxy_data->oob_fd = *((int*) data);
 				g_message("oob fd read from proxy socket %d",
 					*proxy_data->oob_fd);
-				fd_read = true;
 			}
 		}
 	}
 
 	/* only out-of-band data was received */
 	if (bytes_read < sizeof (buf_len) || iov_buffer[0] == 'F' ) {
-		goto out1;
+		proxy_data->oob_fd_found = true;
```

Comment:
```oob_fd_found``` will change to true when **only** out-of-band data is read
I think that this is not equivalent , ```oob_fd_found``` have to change when the fd is copied